Sources/LiveKit/Core/Room.swift

```diff
@@ -345,4 +345,22 @@ extension Room: EngineDelegate {
     func engine(_ engine: Engine, didFailConnection error: Error) {
         notify { $0.room(self, didFailToConnect: error) }
     }
+
+    func engine(_ engine: Engine, didUpdate trackStates: [Livekit_StreamStateInfo]) {
+
+        for trackState in trackStates {
+            // Try to find participant
+            let participant: Participant?
+
+            if trackState.participantSid == localParticipant?.sid {
```

Comment:
You will not get this on local participant.

Reply:
Oh yes so this is fore Remote only